jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java

```diff
@@ -23,167 +23,220 @@
      * @return DataListAssert
      */
     @JDIAction("Assert that '{name}' size {0}")
-    public A size(int size) {
-        return size(equalTo(size));
+    public A size(int size, String... messages) {
+        return size(equalTo(size), messages);
     }
+    
     @JDIAction("Assert that '{name}' size {0}")
-    public A empty() {
-        return size(0);
+    public A empty(String... messages) {
+        return size(0, messages);
     }
+    
     @JDIAction("Assert that '{name}' size {0}")
-    public A notEmpty() {
-        return size(greaterThan(0));
+    public A notEmpty(String... messages) {
+        return size(greaterThan(0), messages);
     }
+    
     @JDIAction("Assert that '{name}' is displayed")
-    public A displayed() {
-        jdiAssert(element.isDisplayed() ? "displayed" : "hidden", Matchers.is("displayed"));
+    public A displayed(String... messages) {
+        jdiAssert(element.isDisplayed() ? "displayed" : "hidden", Matchers.is("displayed"), messages);
         return (A) this;
     }
+    
     @JDIAction("Assert that '{name}' is expanded")
-    public A expanded() {
-        jdiAssert(element.list().isDisplayed() ? "expanded" : "hidden", Matchers.is("expanded"));
+    public A expanded(String... messages) {
+        jdiAssert(element.list().isDisplayed() ? "expanded" : "hidden", Matchers.is("expanded"), messages);
         return (A) this;
     }
+    
     @JDIAction("Assert that '{name}' is collapsed")
-    public A collapsed() {
-        jdiAssert(element.list().isHidden() ? "collapsed" : "expanded", Matchers.is("expanded"));
+    public A collapsed(String... messages) {
+        jdiAssert(element.list().isHidden() ? "collapsed" : "expanded", Matchers.is("expanded"), messages);
         return (A) this;
     }
+    
     @JDIAction("Assert that '{name}' is disappeared")
-    public A disappear() {
-        jdiAssert(element.isHidden() ? "hidden" : "displayed", Matchers.is("hidden"));
+    public A disappear(String... messages) {
+        jdiAssert(element.isHidden() ? "hidden" : "displayed", Matchers.is("hidden"), messages);
         return (A) this;
     }
-
+    
     @JDIAction("Assert that '{name}' is hidden")
-    public A hidden() {
-        return disappear();
+    public A hidden(String... messages) {
+        return disappear(messages);
     }
-    public A notAppear() {
-        return notAppear(TIMEOUT.get());
+    
+    public A notAppear(String... messages) {
+        return notAppear(TIMEOUT.get(), messages);
     }
-
+    
     @JDIAction(value = "Assert that '{name}' does not appear during {0} seconds", timeout = 0)
-    public A notAppear(int timeoutSec) {
+    public A notAppear(int timeoutSec, String... messages) {
         boolean result = new Timer(timeoutSec * 1000)
                 .wait(() -> element.isDisplayed());
-        jdiAssert(result ? "displayed" : "hidden", Matchers.is("hidden"));
+        jdiAssert(result ? "displayed" : "hidden", Matchers.is("hidden"), messages);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5133577)